src/main/scala/com/datastax/sparkstress/DataGenerator.scala

```diff
@@ -15,79 +17,122 @@ abstract class RowGenerator[T] extends Serializable{
+    ss.sparkContext.parallelize(Seq[Int](), numPartitions)
+      .mapPartitionsWithIndex { case (index, n) => generateWideRowByPartitionPartition(index, numPartitions, numTotalKeys, numTotalOps) }
+  }
 
-      for ( pk <- (0L until keysPerPartition).iterator; ck <- (0L until ckeysPerPkey).iterator) yield
-        new WideRowClass((start + pk), (ck).toString, r.nextString(20), r.nextString(20)) 
-    }
+  def getWideRowByPartitionDataset(ss: SparkSession, numPartitions: Int, numTotalOps: Long, numTotalKeys: Long): org.apache.spark.sql.Dataset[WideRowClass] = {
+    import ss.implicits._
+
+    ss.sparkContext.parallelize(Seq[Int](), numPartitions)
+      .mapPartitionsWithIndex { case (index, n) => generateWideRowByPartitionPartition(index, numPartitions, numTotalKeys, numTotalOps) }.toDS()
+  }
 
-    sc.parallelize(Seq[Int](), numPartitions)
-      .mapPartitionsWithIndex { case (index, n) => generatePartition(index) }
+  def generateWideRowPartition(index: Int, numTotalKeys: Long, opsPerPartition: Long) = {
+    val r = new scala.util.Random(index * System.currentTimeMillis())
+    val start = opsPerPartition*index:Long
+    (0L until opsPerPartition).map { i =>
+      new WideRowClass((i + start) % numTotalKeys, (i + start).toString, r.nextString(20), r.nextString(20))
+    }.iterator
   }
 
-  def getWideRowRdd(sc: SparkContext, numPartitions: Int, numTotalOps: Long, numTotalKeys: Long):
+  def getWideRowRdd(ss: SparkSession, numPartitions: Int, numTotalOps: Long, numTotalKeys: Long):
   RDD[WideRowClass] = {
     val opsPerPartition = numTotalOps /numPartitions
 
-    def generatePartition(index: Int) = {
-      val r = new scala.util.Random(index * System.currentTimeMillis())
-      val start = opsPerPartition*index:Long
-      (0L until opsPerPartition).map { i =>
-        new WideRowClass((i + start) % numTotalKeys, (i + start).toString, r.nextString(20), r.nextString(20)) 
-      }.iterator
-    }
-     sc.parallelize(Seq[Int](), numPartitions).mapPartitionsWithIndex {
+     ss.sparkContext.parallelize(Seq[Int](), numPartitions).mapPartitionsWithIndex {
       case (index, n) => {
-        generatePartition(index)
+        generateWideRowPartition(index, numTotalKeys, opsPerPartition)
       }
     }
   }
 
-  def getRandomWideRow(sc: SparkContext, numPartitions: Int, numTotalOps: Long, numTotalKeys:
+  def getWideRowDataset(ss: SparkSession, numPartitions: Int, numTotalOps: Long, numTotalKeys: Long): org.apache.spark.sql.Dataset[WideRowClass] = {
+    import ss.implicits._
+
+    val opsPerPartition = numTotalOps /numPartitions
+
+    ss.sparkContext.parallelize(Seq[Int](), numPartitions).mapPartitionsWithIndex {
+      case (index, n) => {
+        generateWideRowPartition(index, numTotalKeys, opsPerPartition)
+      }
+    }.toDS()
+  }
+
+  def generateRandomWideRowPartition(index: Int, numTotalKeys: Long, opsPerPartition: Long) = {
+    val r = new scala.util.Random(index * System.currentTimeMillis())
```

Comment:
My guess is that Russ seeded this originally in order to control the randomness of the values created. By making it dependent on the current millisecond output we are increasing the randomness of the output generated.